### PR TITLE
Feat/positional encoding

### PR DESCRIPTION
**Pull request by G00sya**
**Closes**

I hope you have a great time reviewing 😸

### CODE REVIEW

File: src/model/positional_encoding.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        super().__init__()
+        self.dropout = nn.Dropout(p=dropout)
+
+        pe = torch.zeros(max_len, d_model)
+        position = torch.arange(0, max_len, dtype=torch.float).unsqueeze(1)
+        div_term = torch.exp(torch.arange(0, d_model, 2, dtype=torch.float) * (-math.log(10000.0) / d_model))
+        pe[:, 0::2] = torch.sin(position * div_term)
+        pe[:, 1::2] = torch.cos(position * div_term)
+        pe = pe.unsqueeze(0)
+        self.register_buffer("pe", pe)
```
Comment (nuryyevva) [inline src/model/positional_encoding.py:25]: Может сюда комментариев добавить о том, что какая строка что делает? Или хотя бы общий смысл, чтобы было понятно что тут творится.

File: tests/model/test_positional_encoding.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
+import torch
+import torch.nn as nn
+
+from src.model.positional_encoding import PositionalEncoding
+
+
```
Comment (nuryyevva) [inline tests/model/test_positional_encoding.py:6]: Может тоже создать фикстуру для:
```
        d_model = 512
        dropout = 0.1
        max_len = 100
        batch_size = 32
        seq_len = 50

        pe = PositionalEncoding(d_model, dropout, max_len)
        inputs = torch.randn(batch_size, seq_len, d_model)
```
чтобы не повторять одно и то же в каждом тесте?

**Review by nuryyevva — CHANGES_REQUESTED** [review]

**Comment by nuryyevva** [comment]
Замечания из другого пуллреквеста про приватные поля и проверки типов передаваемых аргументов и вызова эррора сюда тоже относятся)

File: tests/model/test_positional_encoding.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    def test_forward(self, init_positional_encoding) -> None:
+        """
+        Test the forward pass of the PositionalEncoding module.
+        Checks if the output tensor has the correct shape and if the positional encodings are added correctly.
+        """
+        pe, d_model, dropout, max_len = init_positional_encoding
+        batch_size = 32
+        seq_len = 50
+
+        inputs = torch.randn(batch_size, seq_len, d_model)
```
Comment (nuryyevva) [inline tests/model/test_positional_encoding.py:33]: Тут еще одно замечание и с меня будет аппрув.

Я бы в `init_positional_encoding` добавила:
```
batch_size = 32
seq_len = 50

inputs = torch.randn(batch_size, seq_len, d_model)
```
Потому что они повторяются и в тесте `test_dropout_effect`.

**Review by nuryyevva — CHANGES_REQUESTED** [review]

File: tests/model/test_positional_encoding.py (hunk 3 above)
Comment (G00sya) [reply tests/model/test_positional_encoding.py]: Well done

**Review by nuryyevva — APPROVED** [review]

File: src/model/positional_encoding.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        pe[:, 0::2] = torch.sin(position * div_term)
+
+        # Calculate cosine values for odd columns
+        pe[:, 1::2] = torch.cos(position * div_term)
+
+        # Add a dummy dimension (batch dimension) to the beginning
+        pe = pe.unsqueeze(0)
+
+        # Register the pe tensor as a buffer (not a model parameter)
+        self.register_buffer("pe", pe)
```
Comment (nuryyevva) [inline src/model/positional_encoding.py:48]: `self.dropout` и `self.register_buffer` должны быть с двумя подчёркиваниями

**Review by nuryyevva — CHANGES_REQUESTED** [review]

**Review by ShatunovEvgeniy — CHANGES_REQUESTED** [review]
Всё супер, только для единообразия с остальным котом сделать приватные поля класса

**Review by nuryyevva — APPROVED** [review]
